app/_hub/kong-inc/opa/index.md

```diff
@@ -0,0 +1,307 @@
+      default: "`http`"
+      description: |
+        The protocol to use when talking to Open Policy Agent (OPA) server. Allowed protocols are `http` and `https`.
+    - name: opa_host
+      required: false
+      value_in_examples: localhost
+      datatype: string
+      default: "`localhost`"
+      description: |
+        The DNS name or IP address of the OPA server.
+    - name: opa_port
+      required: false
+      value_in_examples: 8181
+      datatype: integer
+      default: "`8181`"
+      description: |
+        The port of the OPA server.
+    - name: opa_path
+      required: true
+      value_in_examples: </v1/data/example/kong/allow>
+      datatype: string
+      description: |
+        The HTTP path to use when making a request to the OPA server. This is usually the path to the policy and rule to evaluate, prefixed with `/v1/data/`. For example,
+        if you want to evaluate the `allow` rule inside `example.kong` package, then the path would be `/v1/data/example/kong/allow`.
+    - name: include_service_in_opa_input
+      required: false
+      datatype: boolean
+      default: false
+      description: |
+        If set to true, the Kong Service object in use for the current request is included as input to OPA.
+    - name: include_route_in_opa_input
+      required: false
+      datatype: boolean
+      default: false
+      description: |
+        If set to true, the Kong Route object in use for the current request is included as input to OPA.
+    - name: include_consumer_in_opa_input
+      required: false
+      datatype: boolean
+      default: false
+      description: |
+        If set to true, the Kong Consumer object in use for the current request (if any) is included as input to OPA.
+---
+
+## Usage
+
+### Set up an OPA server
+
+If you are not already running an OPA server, start one by following instructions
+on [openpolicyagent.org](https://openpolicyagent.org).
+
+### Set up a dummy policy
+
+Create an `example.rego` file with the following content:
+
+```rego
+package example
+
+default allow = false
+
+allowBoolean {
+  header_present
+}
+
+header_present {
+  input.request.http.headers["my-secret-header"] == "open-sesame"
+}
+
+allowDetailed = response {
+  header_present
+  response := {
+    "allow": true,
+    "headers": {
+      "header-from-opa": "accepted",
+    },
+  }
+}
+
+allowDetailed = response {
+  not header_present
+  response := {
+    "allow": false,
+    "status": 418,
+    "headers": {
+      "header-from-opa": "rejected",
+    },
+  }
+}
+```
+
+This policy allows the request to pass through if it contains the HTTP header
+`my-secret-header` with value `open-sesame`.
+
+
+Next, configure the policy with OPA:
+
+```bash
+curl -XPUT localhost:8181/v1/policies/example --data-binary @example.rego
+```
+
+The above command uses OPA's default port 8181. It could be different for your
+setup.
+
+### Set up Kong
+
+Set up a Route and Service in Kong and then enable the plugin:
```

Comment:
This topic uses "Kong" a lot when it means "Kong Gateway". I'd update every instance of Kong to switch to Kong Gateway - we're not very consistent about this, but we shouldn't contribute to it.

Reply:
> This topic uses "Kong" a lot when it means "Kong Gateway". I'd update every instance of Kong to switch to Kong Gateway - we're not very consistent about this, but we shouldn't contribute to it.

I think we need some shorter variables like KGE for Kong Gateway Enterprise.